src/main/java/com/adobe/cq/commerce/graphql/client/impl/GraphqlClientAdapterFactory.java

```diff
@@ -43,16 +48,26 @@
         AdapterFactory.ADAPTER_CLASSES + "=" + GraphqlClientAdapterFactory.GRAPHQLCLIENT_CLASS_NAME })
 public class GraphqlClientAdapterFactory implements AdapterFactory {
 
+    private static final String CONFIGURATION_SUBSERVICE = "graphql-client-configuration";
+    private static final Logger LOGGER = LoggerFactory.getLogger(GraphqlClientAdapterFactory.class);
+
     protected static final String RESOURCE_CLASS_NAME = "org.apache.sling.api.resource.Resource";
 
     protected static final String GRAPHQLCLIENT_CLASS_NAME = "com.adobe.cq.commerce.graphql.client.GraphqlClient";
 
     protected static final String CONFIG_NAME = "cloudconfigs/commerce";
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(GraphqlClientAdapterFactory.class);
-
     protected Map<String, GraphqlClient> clients = new ConcurrentHashMap<>();
 
+    @Reference
+    private ServiceUserMapped serviceUserMapped;
```

Comment:
`ServiceUserMapped` is a marker service to prevent this component to activate when there's no service user mapped to the bundle. We use it in the connector as well.